File: 242-valid-anagram/valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.length() != t.length()) return false;
        map<char, int> count;

        for (int i = 0; i < s.length(); ++i){
            count[s[i]]++;
            count[t[i]]--;
        }

        for (auto counts : count)
            if (counts.second) return false;
        return true;
    }
};

File: 242-valid-anagram/valid-anagram.py
class Solution(object):
    def checkFreq(self, str):
        freq = {}
        for s in str:
            if s not in freq:
                freq[s] = 1
            else:
                freq[s] += 1
        return freq
    def isAnagram(self, s, t):
        if len(s) != len(t):
            return False
        c1 = self.checkFreq(s)
        c2 = self.checkFreq(t)
        if c1 == c2:
            return True
        return False
        
        